src/gil.rs

```diff
@@ -250,6 +284,20 @@ impl GILGuard {
     }
 }
 
+/// Increment pyo3's internal GIL count - to be called whenever GILPool or GILGuard is created.
+fn increment_gil_count() {
+    GIL_COUNT.with(|c| c.set(c.get() + 1))
+}
+
+/// Decrement pyo3's internal GIL count - to be called whenever GILPool or GILGuard is dropped.
+fn decrement_gil_count() {
+    GIL_COUNT.with(|c| {
+        let current = c.get();
+        assert!(current > 0);
```

Comment:
```suggestion
        debug_assert!(
            current > 0,
            "Negative GIL count is detected. Please report this error to PyO3 repo as a bug."
        );
```

Or any other nice error message.